bergson/score/score.py

```diff
@@ -187,15 +187,20 @@ def get_query_ds(score_cfg: ScoreConfig, device: str, rank: int | None = None):
             if (q and i)
             else (q or i)
         )
-        mixed_preconditioner = {
-            k: v.to(device) for k, v in mixed_preconditioner.items()
-        }
+
+        # Compute H^(-1) via eigendecomposition and apply to query gradients
+        h_inv = {}
+        for name, H in mixed_preconditioner.items():
+            H = H.to(device=device, dtype=torch.float64)
+            H = H + 1e-4 * torch.eye(H.shape[0], device=H.device, dtype=H.dtype)
```

Comment:
@LouisYRYJ I'm planning to merge this but let me know if you'd like a damping value other than 1e-4, or if it should be configurable